docs/metrics/customizing-the-sdk/Program.cs

```diff
@@ -0,0 +1,104 @@
+// <copyright file="Program.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics.Metrics;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+
+public class Program
+{
+    private static readonly Meter MyMeter = new Meter("MyCompany.MyProduct.MyLibrary", "1.0");
+
+    public static void Main(string[] args)
+    {
+        using var meterProvider = Sdk.CreateMeterProviderBuilder()
+            .AddSource(MyMeter.Name)
+            .AddView((instrument) =>
+            {
+                if (instrument.GetType().Name.Contains("Histogram"))
+                {
+                    var metricConfig = new MetricStreamConfig();
+                    metricConfig.HistogramBounds = new double[] { 0, 10, 20 };
+                    return metricConfig;
+                }
+
+                // Null to indicate this View doesn't
+                // have any say on if or how the instrument
+                // gets processed. SDK will do the default here.
+                return null;
+            })
+            .AddView((instrument) =>
+            {
+                if (instrument.GetType().Name.Contains("Histogram"))
+                {
+                    var metricConfig = new MetricStreamConfig();
+                    metricConfig.HistogramBounds = new double[] { 100, 1000 };
+                    metricConfig.Name = "customBucketHistogram";
+                    return metricConfig;
+                }
+
+                return null;
+            })
+            .AddView((instrument) =>
+            {
+                if (instrument.Name.Equals("MyCounter"))
+                {
+                    var metricConfig = new MetricStreamConfig();
+                    metricConfig.Name = "MyCounterRenamed";
```

Comment:
simple rename of an instrument

Reply:
I think this is too complex.
Perhaps we have have overload functions, with simple ones covering 80% of the scenario, and a general purpose one that covers the rest, so in total we cover 100%.

Reply:
yes! the full blown Func callbacks would be good for covering the whole, but easier overloads can cover common scenarios.! 
Will update with some ideas.